src/main/java/edu/harvard/iq/dataverse/sitemap/SiteMapUtil.java

```diff
@@ -0,0 +1,157 @@
+package edu.harvard.iq.dataverse.sitemap;
+
+import edu.harvard.iq.dataverse.Dataset;
+import edu.harvard.iq.dataverse.Dataverse;
+import edu.harvard.iq.dataverse.DvObjectContainer;
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import java.io.File;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.logging.Logger;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+public class SiteMapUtil {
+
+    private static final Logger logger = Logger.getLogger(SiteMapUtil.class.getCanonicalName());
+
+    static final String SITEMAP_FILENAME = "sitemap.xml";
+
+    /**
+     * TODO: Handle more than 50,000 entries in the sitemap.
+     *
+     * (As of this writing Harvard Dataverse only has ~3000 dataverses and
+     * ~30,000 datasets.)
+     *
+     * "each Sitemap file that you provide must have no more than 50,000 URLs"
+     * https://www.sitemaps.org/protocol.html
+     *
+     * Consider using a third party library: "One sitemap can contain a maximum
+     * of 50,000 URLs. (Some sitemaps, like Google News sitemaps, can contain
+     * only 1,000 URLs.) If you need to put more URLs than that in a sitemap,
+     * you'll have to use a sitemap index file. Fortunately, WebSitemapGenerator
+     * can manage the whole thing for you."
+     * https://github.com/dfabulich/sitemapgen4j
+     */
+    public static void updateSiteMap(List<Dataverse> dataverses, List<Dataset> datasets) {
+
+        String sitemapPath = "/tmp";
+        String sitemapPathAndFile;
+        // i.e. /usr/local/glassfish4/glassfish/domains/domain1
+        String domainRoot = System.getProperty("com.sun.aas.instanceRoot");
+        if (domainRoot != null) {
+            // Note that we write to a directory called "sitemap" but we serve just "/sitemap.xml" using PrettyFaces.
+            sitemapPath = domainRoot + File.separator + "docroot" + File.separator + "sitemap";
+        }
+        sitemapPathAndFile = sitemapPath + File.separator + SITEMAP_FILENAME;
+
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
```

Comment:
It seems like this generation could benefit from schema validation via DocumentBuilderFactory.setSchema(schema) ?